sections/faqs/when-to-use-attrs.js

```diff
@@ -0,0 +1,24 @@
+import md from 'components/md'
+
+const WhenToUseAttrs = () => md`
+  ## When to use attrs?
+
+  You can pass in attributes to styled components using \`attrs\`, but
+  it is not always sensible to do so.
+
+  The rule of thumb is to use \`attrs\` when you want every instance of a styled
+  component to have that prop, and pass props directly when every instance needs a
+  different one:
+
+  \`\`\`js
+  const PasswordInput = styled.input.attrs({
+    // Every <PasswordInput /> should be type="password"
+    type: 'password'
+  })\`\`
+
+  // This specific one is hidden, so let's set aria-hidden
+  <PasswordInput aria-hidden="true" />
```

Comment:
Maybe we can also add a paragraph on the lines of:

> The same goes for props that will be inferred based on the “mode” or another prop that are being passed. In this case you can set a property on `attrs` to a function that computes that prop based on other props.

Reply:
Ah, so you mean if you have a prop that's calculated based on another prop, you can do it in `atts` like this?
```js
const MyComponent = styled(SomeComponent).attrs({
  param: props => props.someOtherParam ? 1 : 0
)}``;
```

I can't think of a nice simple real life example of this at the moment. Can you? 🤔

Reply:
Yep, it'd be nice if it's mentioned.

I think a real life example would be `<Button disabled onClick={x} />` since one could override `onClick` when disabled is set to true. But I do not think this needs an example :wink:

Reply:
Fair enough. I've added a paragraph on this as requested, thanks Phil ✨